### ISSUE TITLE
focus (scroll) to targetViews that are not visible

### ISSUE THREAD

**Issue by lenoch0d** [KeepSafe/TapTargetView#111]
**Version used:** 1.7.0
**Android version:** 6.0

Hello.

In cases, when views are put into (Nested)ScrollView, and as a TapTargetView a view which is currently ivisible on the screen is set, the Target becomes untouchable (as it was created  "below the screen"). It would be nice if the Target was focused (=scrollView scrolled to target position) before creating the TapTargetView.

I've tried to implement it by myself in onSequenceStep() method - by calling requestChildFocus(target,target). It scrolls/focuses fine,  but the TapTargetView position differs from the actual targetView position.

Any ideas how to solve it? Thank you!

**Comment by xiphirx** [comment]
The issue there is that your TapTargets are being created at the same time as the sequence. This means that you will need to delay the creation of your TapTargets until they're about to be shown. You can either do this by using multiple sequences that are strung together by a call like requesFocus or your own custom implementation of a TapTargetSequence. I'm not sure this is something the library should do at the moment.

**Comment by shivamsoods** [comment]
Hey
I also faced a similar problem so what I did was before starting the tap sequence I checked whether the concerned view was visible to user or not. I knew that at most 3 items will always be on screen, so I made  a sequence for that.

Then in the  onSequenceFinish() I requested the focus to my view which might or might not be visible to user.

Kotlin Code:

```                  
your_view.requestChildFocus(childView,childView)
```

I had created another sequence for the views that might not be visible to the user.
So to summarize, create two sequences, one for the views that will definitely visible to the user and another for the views that might/will not be visible to user. After the sequence has ended for the  definitely visible views, request the focus for the non visible views to user and start the second sequence.
